controller/haproxy/rules.go

```diff
@@ -66,7 +66,7 @@ var constLookup = map[RuleType]string{
 type RuleStatus int
 
 // RuleID uniquely identify a HAProxy Rule
-type RuleID uint32
+type RuleID string
```

Comment:
Changing this requires also updating this line https://github.com/haproxytech/kubernetes-ingress/blob/ebcd3ce6c37494076b877fcaca033368443b671c/controller/haproxy/rules.go#L187
Otherwise it will result in incorrect conversion and buggy haproxy rules selection